### PR TITLE
PDCT-716/Make-library-for-alembic-db-schema

### PR DESCRIPTION
**Pull request by asacristani**
# Description

## DONE
1. Remove client/db folder
2. Start to use [navigator-db-client](https://github.com/climatepolicyradar/navigator-db-client/tree/v.0.0.1) library
3. Adapt code (including tests)
4. Move tests (no integrity) that affects only to the library -> NO TEST FOUND
5. Check the compatibility when navigator-backend will be adapted

## TO DISCUSS
- pip/poetry in Dockers -> Poetry
- RepositoryError -> Errors stay in backends

## Type of change

Please select the option(s) below that are most relevant:

- [ ] Bug fix
- [ ] New feature
- [x] Breaking change

## How Has This Been Tested?

Unit and integration.
Staging test before Production.

## Reviewer Checklist

- [ ] The PR represents a single feature (small drive-by fixes are also ok)
- [x] The PR includes tests that are sufficient for the level of risk
- [x] The code is sufficiently commented, particularly in hard-to-understand areas
- [x] Any required documentation updates have been made
- [x] Any TODOs added are captured in future tickets
- [x] No FIXMEs remain

### CODE REVIEW

File: app/api/api_v1/routers/analytics.py [hunk 1]
```diff
@@ -1,7 +1,7 @@
 """Endpoints for managing the Analytics service."""
 import logging
 from fastapi import APIRouter, HTTPException, status
-from app.errors import RepositoryError
+from app.clients.db.errors import RepositoryError
```
Comment (diversemix) [inline app/api/api_v1/routers/analytics.py:4]: Hmm... I thought we were leaving the `RepositoryError` in here? Why did we move it again?

File: pyproject.toml [hunk 2]
```diff
@@ -28,6 +28,7 @@ boto3 = "^1.28.46"
 moto = "^4.2.2"
 types-sqlalchemy = "^1.4.53.38"
 urllib3 = "^1.26.17"
+db-client = {git = "https://github.com/climatepolicyradar/navigator-db-client.git", tag = "main-bd82dba6"}
```
Comment (diversemix) [inline pyproject.toml:31]: Can we use a semver tag please :pray:

File: app/api/api_v1/routers/analytics.py (hunk 1 above)
Comment (asacristani) [reply app/api/api_v1/routers/analytics.py]: Reverted!

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -36,15 +36,15 @@ jobs:
         run: |
           poetry run pre-commit run --all-files
 
+      - name: Use .env.example
```
Comment (katybaulch) [inline .github/workflows/ci.yml:39]: Not sure this does anything? We aren't telling the make build command to use the env file, so this is redundant

File: Dockerfile [hunk 4]
```diff
@@ -1,13 +1,12 @@
 FROM python:3.9-slim
 
 WORKDIR /usr/src
-ENV PYTHONPATH=/usr/src
+ENV PYTHONPATH=/usr/src
```
Comment (katybaulch) [inline Dockerfile:4]: Trailing whitespace

File: app/config.py [hunk 5]
```diff
@@ -9,4 +9,8 @@
 SQLALCHEMY_DATABASE_URI = (
     f"postgresql://{_creds}@{ADMIN_POSTGRES_HOST}:5432/{ADMIN_POSTGRES_DATABASE}"
 )
+
+# This used by db-client and alembic script for migrations
+os.environ["DATABASE_URL"] = SQLALCHEMY_DATABASE_URI
```
Comment (katybaulch) [inline app/config.py:14]: Should we be setting this in the script, or in the environment we are running the admin service from?

File: docker-compose.yml [hunk 6]
```diff
@@ -0,0 +1,38 @@
+version: '3.7'
```
Comment (katybaulch) [inline docker-compose.yml:1]: What's the rationale behind including a docker-compose file here now that we've moved to have a single source of truth by combining the admin service and backend databases?

File: app/initial_data.py [hunk 7]
```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env python3
```
Comment (katybaulch) [inline app/initial_data.py:1]: See comment on docker-compose

File: app/config.py (hunk 5 above)
Comment (diversemix) [reply app/config.py:14]: I think the idea is to build all the necessary config from the `ADMIN_...` vars - unfortunately the only way of communicating this down to the `db-client` library seems to be setting the env var `DATABASE_URL` ... the `db-client` has a function `get_url()` which returns this value.

Comment (diversemix) [reply app/config.py:14]: So you can see from line 10 that the environment specifies the ADMIN vars and the code worries about formatting it up for the rest of the code. This make sense @katybaulch

File: docker-compose.yml (hunk 6 above)
Comment (diversemix) [reply docker-compose.yml:1]: I guess to fully de-couple the two things and also letting them develop independently. 

For example: If the nav-backend breaks the ability to spin up a local db - then admin development can still continue. Arguably - we could move this into the `db-client` in some way in the future. But for now I think this is the simplest/less risky approach.

Happy to discuss further.

File: app/initial_data.py (hunk 7 above)
Comment (diversemix) [reply app/initial_data.py:1]: See https://github.com/climatepolicyradar/navigator-admin-backend/pull/62#discussion_r1489345597 :smile:

**Review by diversemix — APPROVED** [review]
I'm happy - as long as we get another review :grin:

File: app/config.py (hunk 5 above)
Comment (katybaulch) [reply app/config.py:14]: Ah understood, thanks for clarifying

**Review by katybaulch — APPROVED** [review]
All good, well done! 🦖

**Pull request merged**